cmd/acra-key-tool/acra-key-tool.go

```diff
@@ -154,18 +194,61 @@ func main() {
 		if err != nil {
 			log.WithError(err).Fatal("Cannot read zone storage private key")
 		}
-		keyBytes = key.Value
+		return key.Value
 	default:
-		log.Fatalf("Unknown key kind: %v, allowed values: %s", params.KeyKind, keyKinds)
+		log.Fatalf("Unknown key kind: %v, allowed values: %s", params.ReadKeyKind, readKeyKinds)
+		return nil
 	}
+}
 
-	_, err = os.Stdout.Write(keyBytes)
+func destroyKey() {
+	keyStore, err := openKeyStoreForModification()
 	if err != nil {
-		log.Fatalf("Failed to write key bytes: %v", err)
+		log.Fatalf("Failed to open key store: %v", err)
+	}
+
+	switch params.DestroyKeyKind {
+	case keyTransportConnector:
+		if params.ClientID == "" {
+			log.Fatal("--destroy_key " + keyTransportConnector + " requires --client_id")
+		}
+		err := keyStore.DestroyConnectorKeypair([]byte(params.ClientID))
+		if err != nil {
+			log.WithError(err).Fatal("Cannot destroy AcraConnector transport key pair")
+		}
+	case keyTransportServer:
+		if params.ClientID == "" {
+			log.Fatal("--destroy_key " + keyTransportServer + " requires --client_id")
+		}
+		err := keyStore.DestroyServerKeypair([]byte(params.ClientID))
+		if err != nil {
+			log.WithError(err).Fatal("Cannot destroy AcraServer transport key pair")
+		}
+	case keyTransportTranslator:
+		if params.ClientID == "" {
+			log.Fatal("--destroy_key " + keyTransportTranslator + " requires --client_id")
+		}
+		err := keyStore.DestroyTranslatorKeypair([]byte(params.ClientID))
+		if err != nil {
+			log.WithError(err).Fatal("Cannot destroy AcraTranslator transport key pair")
+		}
+	default:
+		log.Fatalf("Unknown key kind: %v, allowed values: %s", params.DestroyKeyKind, destroyKeyKinds)
+	}
+}
+
+func openKeyStoreForReading() (keystore.ServerKeyStore, error) {
+	switch params.KeyStoreVersion {
+	case "v1":
+		return openKeyStoreV1()
+	case "v2":
+		return openKeyStoreV2()
+	default:
+		return nil, fmt.Errorf("unknown keystore option: %v", params.KeyStoreVersion)
```

Comment:
We don't really need to be specific here since this error is going to be reported like this:

https://github.com/cossacklabs/acra/blob/11a15c019c73426a9152e64e3fdd2d5fe2c9e497/cmd/acra-read-key/acra-read-key.go#L101-L104

But changing “option” to “version” is something that we'd probably want to do. Initially I meant the `--keystore` *command-line option* but the person looking at this error will probably that. And this wording will probably work better after we remove `v1` support and make it an error. (Note to self: other services may use this line too, they need to be updated as well.)